types/jest/index.d.ts

```diff
@@ -695,7 +695,7 @@ declare namespace jest {
          *   expect(mock).toBeCalledWith(expect.any(Number));
          * });
          */
-        any<T extends Constructor>(classType: T): T extends Func ? ReturnType<T> : InstanceType<T>;
+        any<T>(classType: T): T extends Func ? ReturnType<T> : T extends Constructor ? InstanceType<T> : any;
```

Comment:
I still think it should be reverted back to `any`. Or to be typed with its real type.

```suggestion
        any(classType: any): any;
```